File: nodejs/base/core/lib/logging/console-logger-provider.ts
import { LoggerProvider } from "./logger-provider";
import { Logger } from "./logger";
import { McmaTrackerProperties } from "../model";
import { ConsoleLogger } from "./console-logger";

export class ConsoleLoggerProvider implements LoggerProvider {
    constructor(private source: string) { }

    get(requestId?: string, tracker?: McmaTrackerProperties): Logger {
        return new ConsoleLogger(this.source, requestId, tracker);
    }
}

File: dotnet/base/Mcma.Core/Logging/ConsoleLoggerProvider.cs
namespace Mcma.Logging
{
    public class ConsoleLoggerProvider : ILoggerProvider
    {
        public ConsoleLoggerProvider(string source)
        {
            Source = source;
        }

        private string Source { get; }

        public ILogger Get(string requestId = null, McmaTracker tracker = null)
            => new ConsoleLogger(Source, requestId, tracker);
    }
}